ntbs-service/Pages/Index.cshtml.cs

```diff
@@ -41,6 +42,8 @@ public class IndexModel : PageModel
         
         public async Task OnGetAsync()
         {
+            HttpContext.Session.SetString("LastVisitedPageTitle", "Index");
+            HttpContext.Session.SetString("LastVisitedPageUrl", "/Index");
```

Comment:
Given our conversation about this, I don't think "LastVisited" quite captures what these are storing. Not sure I have a perfect suggestion, but perhaps it's something along the lines of "TopLevelLocation" or "BreadcrumbsLocation" or similar?